loopy/schedule/__init__.py

```diff
@@ -166,32 +176,107 @@ def get_insn_ids_for_block_at(schedule, start_idx):
             if isinstance(sub_sched_item, RunInstruction))
 
 
-def find_active_inames_at(kernel, sched_index):
+def find_active_inames_at(kernel, sched_indices):
```

Comment:
Likewise, I think this is begging for the type of rapid query object I request above.

On a semi-related note, @jdsteve2 restructured some of his code to avoid quadratic run time earlier for (IIRC) the same query... it could have stayed the way it was if we had had this query.